tests/conftest.py

```diff
@@ -77,6 +77,15 @@ def client(request):
     if request.node.get_closest_marker("skip_t1") and client.features.model == "1":
         pytest.skip("Test excluded on Trezor 1")
 
+    client.open()
+    debug_state = client.debug.state()
+    client.close()
+    if request.node.get_closest_marker("sd_card"):
+        if client.features.model == "1":
+            pytest.skip("SD card test skipped on T1")
+        if not debug_state.sd_card:
+            pytest.skip("SD card not present")
```

Comment:
I would still like to see a warning emitted for each test that gets skipped due to the SD card not being present:
`warnings.warn("SD card not present. The {} was skipped.".format(test_name))`
Consider for example when we are running tests on the device before releasing new firmware. If we forget to insert the card then the tests will get silently skipped without us ever realizing it. Furthermore the SD card related tests are precisely the ones that you _really_ want to run on the device, because I bet half of the SD card code on the emulator is different.

Reply:
warning is too weak a mechanism for this
if we actually want to enforce this, I propose we hard-fail the tests if, say, a given command-line option is supplied. or not supplied.

...as it happens, without any special handling on our side, `pytest -m "not sd_card"` will skip sdcard tests. Then we can simply not test whether we have SD card, and let the tests fail if they actually require sd card